### PR TITLE
Fix bug when repository deleted but its issues not

### PR DESCRIPTION
**Pull request by lunny**

### CODE REVIEW

**Comment by guillep2k** [comment]
So this is a paliative measure? Why would the issues remain in the database when the repos have been deleted?

File: models/issue_list.go [hunk 1]
```diff
@@ -415,8 +415,11 @@ func (issues IssueList) loadTotalTrackedTimes(e Engine) (err error) {
 
 	var ids = make([]int64, 0, len(issues))
 	for _, issue := range issues {
-		if issue.Repo.IsTimetrackerEnabled() {
-			ids = append(ids, issue.ID)
+		issue.loadRepo(e)
+		if issue.Repo != nil {
```
Comment (lafriks) [inline models/issue_list.go]: merge both `if` in single condition
```suggestion
		if issue.Repo != nil && issue.Repo.IsTimetrackerEnabled() {
```

**Review by lafriks — CHANGES_REQUESTED** [review]

File: models/issue_list.go (hunk 1 above)
Comment (lunny) [reply models/issue_list.go]: done.

File: models/issue_list.go [hunk 2]
```diff
@@ -415,7 +415,8 @@ func (issues IssueList) loadTotalTrackedTimes(e Engine) (err error) {
 
 	var ids = make([]int64, 0, len(issues))
 	for _, issue := range issues {
-		if issue.Repo.IsTimetrackerEnabled() {
+		issue.loadRepo(e)
```
Comment (lafriks) [inline models/issue_list.go]: drone build fails:
```
models/issue_list.go:418:17: Error return value of `issue.loadRepo` is not checked (errcheck)
```

File: routers/user/home.go [hunk 3]
```diff
@@ -332,11 +332,18 @@ func Issues(ctx *context.Context) {
...
-		if isPullList {
-			commitStatus[issue.PullRequest.ID], _ = issue.PullRequest.GetLastCommitStatus()
+	for i := 0; i < len(issues); {
+		issue := issues[i]
+		repo, ok := showReposMap[issue.RepoID]
+		if ok {
+			issue.Repo = repo
+
+			if isPullList {
+				commitStatus[issue.PullRequest.ID], _ = issue.PullRequest.GetLastCommitStatus()
```
Comment (guillep2k) [inline routers/user/home.go:342]: Is it certain that the error can be ignored?

File: routers/user/home.go [hunk 4]
```diff
@@ -332,11 +332,18 @@ func Issues(ctx *context.Context) {
...
-	for _, issue := range issues {
-		issue.Repo = showReposMap[issue.RepoID]
-
-		if isPullList {
-			commitStatus[issue.PullRequest.ID], _ = issue.PullRequest.GetLastCommitStatus()
+	for i := 0; i < len(issues); {
+		issue := issues[i]
+		repo, ok := showReposMap[issue.RepoID]
+		if ok {
+			issue.Repo = repo
```
Comment (guillep2k) [inline routers/user/home.go:339]:
```suggestion
		issue.Repo, ok := showReposMap[issue.RepoID]
		if ok {
```

**Comment by lunny** [comment]
I think this could be closed. It assumes the data is broken but I think it should not.

**Pull request closed without merging**